app/models/king.rb

```diff
@@ -1,2 +1,11 @@
 class King < Piece
+	def valid_move?(new_rank, new_file)
+		rank_diff = (new_rank - rank).abs
+		file_diff = (new_file - file).abs
+		if (rank_diff <= 1) && (file_diff <= 1) && !is_move_obstructed?(new_rank, new_file) && !is_move_malformed?(new_rank, new_file)
```

Comment:
I don't understand that `is_move_obstructed?` handles malformed moves here. It only covers obstruction moves. `is_move_malformed?` handles malformed moves here.

And do you suggests to do like this?

def valid_move?(new_rank, new_file)
    return false if is_move_malformed?
    return false if is_move_obstructed?
    (new_rank - rank).abs <= 1 && (new_file - file).abs <=1
end